src/Analyzers/Core/CodeFixes/MatchFolderAndNamespace/AbstractChangeNamespaceToMatchFolderCodeFixProvider.cs

```diff
@@ -46,7 +46,7 @@ private static async Task<Solution> FixAllInDocumentAsync(Document document, Imm
             // us to keep in line with the sync methodology that we have as a public API and not have 
             // to rewrite or move the complex logic. RenameDocumentAsync is designed to behave the same
             // as the intent of this analyzer/codefix pair.
-            var targetFolders = PathMetadataUtilities.BuildFoldersFromNamespace(targetNamespace);
+            var targetFolders = PathMetadataUtilities.BuildFoldersFromNamespace(targetNamespace, document.Project.DefaultNamespace);
```

Comment:
The analyzer has already correctly calculated the correct namespace. See the diagnostic message:

![image](https://user-images.githubusercontent.com/31348972/128295793-f011f0ce-2af7-4d99-80e6-ea8b210ec960.png)

Can't we just pass it in the properties bag and avoid recalculating it?

Reply:
We already are. The fix here is to avoid duplicating the default namespace. `BuildFoldersFromNamespace` will build relative folders, but if it doesn't know the default/root namespace it will do it incorrectly. 

See how targetNamespace gets assigned. It comes from the analyzer